### PR TITLE
*/*: fix brace style, Bash tests, eapply

### PR DESCRIPTION
**Pull request by thesamesam**
Mostly brace style fixing but also one instance of POSIX -> Bash tests and epatch -> eapply.

### CODE REVIEW

File: ebuild-writing/functions/pkg_config/text.xml [hunk 1]
```diff
@@ -49,7 +48,7 @@ Taken from the <c>mysql</c> ebuilds. Note the use of <c>${ROOT}</c>.
 
 <codesample lang="ebuild">
 pkg_config() {
-	if [ ! -d "${ROOT}"/var/lib/mysql/mysql ] ; then
+	if [[ ! -d "${ROOT}"/var/lib/mysql/mysql ]] ; then
```
Comment (ulm) [inline ebuild-writing/functions/pkg_config/text.xml:51]: No quotation marks here.

File: ebuild-writing/functions/src_compile/building/text.xml [hunk 2]
```diff
@@ -60,7 +60,7 @@ src_prepare() {
 	# We have a weird Makefile to work with which ignores our
 	# compiler preferences. yay!
 	# Note the single quotes.
-	sed -i -e 's:cc -O2:$(CC) $(CFLAGS) $(LDFLAGS):' Makefile \
+	sed -i -e 's;cc -O2;$(CC) $(CFLAGS) $(LDFLAGS);' Makefile \
```
Comment (ulm) [inline ebuild-writing/functions/src_compile/building/text.xml:63]: Why this change? We should stick to the common separator characters (like `/` or `:`). Unusual ones like `;` only make the reader wonder if there's a reason for it.

**Review by ulm — CHANGES_REQUESTED** [review]

File: ebuild-writing/functions/src_compile/building/text.xml (hunk 2 above)
Comment (thesamesam) [reply ebuild-writing/functions/src_compile/building/text.xml:63]: Did you read the commit message? :)

```
Author: Sam James <sam@gentoo.org>
Date:   Sat Mar 20 09:55:09 2021 +0000

    ebuild-writing/functions/src_compile/building: use ; in CFLAGS sed

    A colon is a valid character in CFLAGS, so we shouldn't use it
    as a sed delimiter.

    Bug: https://bugs.gentoo.org/685160
    Signed-off-by: Sam James <sam@gentoo.org>
```

File: ebuild-writing/functions/pkg_config/text.xml (hunk 1 above)
Comment (thesamesam) [reply ebuild-writing/functions/pkg_config/text.xml:51]: Really? This is common style.

Comment (ulm) [reply ebuild-writing/functions/pkg_config/text.xml:51]: Which doesn't make it right. `[[ ]]` is a syntax element, and no word splitting or globbing will take place. So the quotation marks are excessive (but they were necessary inside `[ ]`).

File: ebuild-writing/functions/src_compile/building/text.xml (hunk 2 above)
Comment (ulm) [reply ebuild-writing/functions/src_compile/building/text.xml:63]: I don't see any colon in `$(CC) $(CFLAGS) $(LDFLAGS)` (and neither in `cc -O2`).

Maybe I need new glasses? :smile:

Comment (ulm) [reply ebuild-writing/functions/src_compile/building/text.xml:63]: Ah, I think I understand now.
Note the single quotes, `$(CFLAGS)` is a literal string here.

Comment (ulm) [reply ebuild-writing/functions/src_compile/building/text.xml:63]: Also, if it wasn't a literal string, then `;` would only marginally be better than `:`. CFLAGS originates from user configuration, so one cannot assume anything about it, and it may contain _any_ character. Expanding the variable in a sed replacement is simply asking for trouble and should be avoided.

Comment (thesamesam) [reply ebuild-writing/functions/src_compile/building/text.xml:63]: Ahhh, I missed it was literal! It's still extremely common to end up sedding CFLAGS but I take the point that it's a rabbit hole.

I'd still say that using something other than a comma or a colon is ideal for where people don't want to fix the build system though. The example in the devmanual should probably be 'less wrong', but maybe it'd be useful to put a note about this as well somewhere?

File: ebuild-writing/functions/pkg_config/text.xml (hunk 1 above)
Comment (thesamesam) [reply ebuild-writing/functions/pkg_config/text.xml:51]: You're right, of course, but I've noticed a common theme between both of the changes I've suggested here.

Both are imperfect (and one is unnecessary) but both _suggest_ better habits, even if imperfect. I'm happy to drop it here though.

File: ebuild-writing/functions/src_compile/building/text.xml (hunk 2 above)
Comment (thesamesam) [reply ebuild-writing/functions/src_compile/building/text.xml:63]: Or maybe just a warning in the sed section about choosing your delimiter?

File: ebuild-writing/functions/pkg_config/text.xml (hunk 1 above)
Comment (ulm) [reply ebuild-writing/functions/pkg_config/text.xml:51]: Yes, please drop the quotes. If you leave them in place, there's little point in using `[[ ]]` instead of `[ ]`.

Of course this is very minor, so I wouldn't file a bug for seeing quotes in `[[ ]]`. But I would comment about it in eclass review.

File: ebuild-writing/functions/src_compile/building/text.xml (hunk 2 above)
Comment (ulm) [reply ebuild-writing/functions/src_compile/building/text.xml:63]: We say something about it [here](https://devmanual.gentoo.org/tools-reference/sed/index.html#simple-text-substitution-using-sed).

File: ebuild-writing/functions/pkg_config/text.xml (hunk 1 above)
Comment (thesamesam) [reply ebuild-writing/functions/pkg_config/text.xml:51]: Understood. Let's fix that.

File: ebuild-writing/functions/src_compile/building/text.xml (hunk 2 above)
Comment (thesamesam) [reply ebuild-writing/functions/src_compile/building/text.xml]: I've added some more information there.

File: ebuild-writing/common-mistakes/text.xml [hunk 3]
```diff
@@ -259,7 +259,7 @@ part.
 </body>
 </subsection>
 <subsection>
-<title>DEPEND is incomplete</title>
+<title>(B/R)DEPEND is incomplete</title>
```
Comment (ulm) [inline ebuild-writing/common-mistakes/text.xml:262]: Is this heading change necessary? It will break external links to the section.

Maybe mention instead in the first section that this applies to all dependency variables?

File: ebuild-writing/functions/src_compile/building/text.xml [hunk 4]
```diff
@@ -59,8 +59,8 @@ src_prepare() {
 
 	# We have a weird Makefile to work with which ignores our
 	# compiler preferences. yay!
-	# Note the single quotes.
-	sed -i -e 's:cc -O2:$(CC) $(CFLAGS) $(LDFLAGS):' Makefile \
+	# Note the single quotes (hence the delimiter is not an issue)
+	sed -i -e 's;cc -O2:$(CC) $(CFLAGS) $(LDFLAGS):' Makefile \
```
Comment (ulm) [inline ebuild-writing/functions/src_compile/building/text.xml:63]: Missed one.

File: general-concepts/dependencies/text.xml [hunk 5]
```diff
@@ -461,6 +461,12 @@ DEPEND="!build? (
 <title>Any of Many Dependencies</title>
 <body>
 
+<note>
+To ease dependency resolution for Portage, it is recommended that you sort
+the atoms in <e>preferred</e> order first. See
```
Comment (ulm) [inline general-concepts/dependencies/text.xml:466]: No "atoms" please. We're trying to abandon this term ever since [bug 174322](https://bugs.gentoo.org/174322) (with mixed success, I must admit).

File: general-concepts/dependencies/text.xml [hunk 6]
```diff
@@ -626,12 +632,13 @@ package:
   </dd>
   <dt>
     Remember non-binary dependencies such as pkg-config, doc generation
-    programs, etc.
+    programs, etc. Such programs would usually belong in <c>BDEPEND</c>
```
Comment (ulm) [inline general-concepts/dependencies/text.xml:635]: Full stop.

File: general-concepts/dependencies/text.xml [hunk 7]
```diff
@@ -626,12 +632,13 @@ package:
...
   </dt>
   <dd>
     Usually the build process requires some dependencies such as intltool,
     libtool, pkg-config, doxygen, scrollkeeper, gtk-doc, etc. Make sure those
-    are clearly stated.
+    are clearly stated. Again, such dependencies usually belong in
+    <c>BDEPEND</c>.
```
Comment (ulm) [inline general-concepts/dependencies/text.xml:641]: Line break is not necessary here.

File: tools-reference/sed/text.xml [hunk 8]
```diff
@@ -65,7 +65,7 @@ constructs are specific to <c>GNU sed 4</c> <d/> on non-GNU userland archs, the
 guaranteed to be installed as part of <c>@system</c>. This was not
 always the case,
 which is why some packages, particularly those which use <c>sed -i</c>, have
-<c>DEPEND</c> s upon <c>&gt;=sys-apps/sed-4</c>.
+<c>BDEPEND</c> s upon <c>&gt;=sys-apps/sed-4</c>.
```
Comment (ulm) [inline tools-reference/sed/text.xml:68]: What's that lonely "s" doing here?

Also, this is about old packages, which most certainly have &gt;=sed-4 in DEPEND, not in BDEPEND.

**Review by ulm — COMMENTED** [review]
While I previously said that combining several small issues in one PR is fine, *adding* unrelated issues to an existing PR definitely isn't, especially after a first round of review has taken place. It makes it a) hard to see what has changed in the existing commits, and b) makes review of the new stuff more difficult.

So please, file a separate PR next time. (Don't spilt this one, though.)

File: tools-reference/sed/text.xml (hunk 8 above)
Comment (thesamesam) [reply tools-reference/sed/text.xml:68]: Oh, duh. Let's fix the s though.

File: general-concepts/dependencies/text.xml (hunk 5 above)
Comment (ulm) [reply general-concepts/dependencies/text.xml:466]: Also, more specifically, [bug 603970](https://bugs.gentoo.org/603970) for the devmanual.

File: ebuild-writing/common-mistakes/text.xml (hunk 3 above)
Comment (thesamesam) [reply ebuild-writing/common-mistakes/text.xml]: Done.

File: general-concepts/dependencies/text.xml (hunk 5 above)
Comment (thesamesam) [reply general-concepts/dependencies/text.xml]: Ah, I should've remembered that. I'm not sure this is realistic until Portage itself doesn't use 'atom' in its output though. That's exactly where I picked it up.

**Review by ulm — APPROVED** [review]

**Comment by ulm** [comment]
I've cherry-picked some trivial things to get them out of the way. Please rebase.

**Comment by ulm** [comment]
Also, can you drop that change of `:` to `;` from commit 55acfbc8995253760d2cd33ca2c6e0311bc9c236, instead of reverting it in a later commit?

**Comment by thesamesam** [comment]
Done.

File: general-concepts/dependencies/text.xml [hunk 9]
```diff
@@ -461,6 +461,12 @@ DEPEND="!build? (
 <title>Any of Many Dependencies</title>
 <body>
 
+<note>
+To ease dependency resolution for Portage, it is recommended that you sort
+the elements in <e>preferred</e> order first. See
+<uri link="https://bugs.gentoo.org/489458">this bug</uri> for more context.
```
Comment (ulm) [inline general-concepts/dependencies/text.xml:467]: I'd prefer "[bug 489458](https://bugs.gentoo.org/489458)" instead of "[this bug](https://bugs.gentoo.org/489458)" here. I'll update this when merging, if you don't mind.

Comment (thesamesam) [reply general-concepts/dependencies/text.xml:467]: Completely fine with me. I'll keep this style preference in mind in future.

**Comment by ulm** [comment]
Thanks, merged.
